### PR TITLE
Fix data race in `TestManagedConn/Basic`

### PR DESCRIPTION
**Pull request by espadolini**
`CloseIdleConnections` will synchronously close all idle connections but the connection we wanted to get closed could've not transitioned to an idle state in the internals of the `http.Transport`. Instead, we set the idle timeout to the lowest possible value and then we wait until the connection gets closed.

### CODE REVIEW

File: lib/resumption/managedconn.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ const (
 // connections actually return ECONNRESET on the first syscall experiencing the
 // error, then EPIPE afterwards; we don't bother emulating that detail and
 // always return EPIPE instead.
-var errBrokenPipe = syscall.EPIPE
+var errBrokenPipe error = syscall.EPIPE
```
Comment (espadolini) [inline lib/resumption/managedconn.go:46]: (this is unrelated to the fix, it's just a minor tweak in the same package)

**Review by rosstimothy — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]
